generatebundlefile/promote.go

```diff
@@ -112,16 +114,31 @@ func (c *SDKClients) PromoteHelmChart(repository, authFile, tag string, copyImag
 	// If we are not moving artifacts to the Private ECR Packages artifact account, get information from public ECR instead.
 	// The first scenario runs for flags --private-profile and --promote.
 	// The 2nd scenario runs for flags and --public-profile.
-	if c.ecrClientRelease != nil || c.ecrClient != nil {
-		name, version, sha, err = c.getNameAndVersion(repository, tag, c.stsClient.AccountID)
-		if err != nil {
-			return fmt.Errorf("Error getting name and version from helmchart %s", err)
+	if regionalBuildMode {
```

Comment:
If running in regional build mode, the helm chart destination is the artifacts account public ECR hence we get the name and version from there, else we get it from private ECR.